### PR TITLE
Add pre-emphasis and de-emphasis functions

### PR DESCRIPTION
**Pull request by hwangjeff**
Adds pre-emphasis and de-emphasis functions.

### CODE REVIEW

**Comment by carolineechen** [comment]
linking some relevant issues: #226 #2323

File: test/torchaudio_unittest/prototype/functional/torchscript_consistency_test_impl.py [hunk 1]
```diff
@@ -85,6 +85,18 @@ def test_sinc_ir(self):
     def test_speed(self):
         leading_dims = (3, 2)
         T = 200
-        waveform = torch.rand(*leading_dims, T, dtype=self.dtype, device=self.device, requires_grad=True)
+        waveform = torch.rand(*leading_dims, T, dtype=self.dtype, device=self.device)
```
Comment (carolineechen) [inline test/torchaudio_unittest/prototype/functional/torchscript_consistency_test_impl.py:88]: was this meant to be modified?

File: test/torchaudio_unittest/prototype/functional/torchscript_consistency_test_impl.py [hunk 2]
```diff
@@ -85,6 +85,18 @@ def test_sinc_ir(self):
...
         lengths = torch.randint(1, T, leading_dims, dtype=self.dtype, device=self.device)
         self._assert_consistency(F.speed, (waveform, lengths, 1000, 1.1))
+
+    def test_preemphasis(self):
+        waveform = torch.rand(3, 2, 100, device=self.device, dtype=self.dtype)
+        coeff = 0.9
+        self._assert_consistency(F.preemphasis, (waveform, coeff))
+        self._assert_consistency(F.preemphasis, (waveform, coeff))
```
Comment (carolineechen) [inline test/torchaudio_unittest/prototype/functional/torchscript_consistency_test_impl.py:96]: duplicate line

File: test/torchaudio_unittest/prototype/functional/torchscript_consistency_test_impl.py [hunk 3]
```diff
@@ -85,6 +85,18 @@ def test_sinc_ir(self):
...
+
+    def test_deemphasis(self):
+        waveform = torch.rand(3, 2, 100, device=self.device, dtype=self.dtype)
+        coeff = 0.9
+        self._assert_consistency(F.deemphasis, (waveform, coeff))
+        self._assert_consistency(F.deemphasis, (waveform, coeff))
```
Comment (carolineechen) [inline test/torchaudio_unittest/prototype/functional/torchscript_consistency_test_impl.py:102]: duplicate line

File: test/torchaudio_unittest/prototype/functional/torchscript_consistency_test_impl.py (hunk 1 above)
Comment (hwangjeff) [reply test/torchaudio_unittest/prototype/functional/torchscript_consistency_test_impl.py]: yeah i realized it wasn't necessary, but i can deal with it in a separate pr

**Review by carolineechen — APPROVED** [review]

File: torchaudio/prototype/functional/functional.py [hunk 4]
```diff
@@ -345,3 +344,45 @@ def speed(
...
+    Args:
+        waveform (torch.Tensor): Waveform, with shape `(..., N)`.
+        coeff (float, optional): Pre-emphasis coefficient. Typically between 0.0 and 1.0.
+            (Default: 0.97)
+
+    Returns:
+        torch.Tensor: Pre-emphasized waveform, with shape `(..., N)`.
+    """
+    waveform = waveform.clone()
+    waveform[..., 1:] -= coeff * waveform[..., :-1]
```
Comment (hwangjeff) [inline torchaudio/prototype/functional/functional.py:366]: note that i opted for this simplified implementation to bypass lfilter — it is significantly faster

**Pull request closed without merging**